Description: Add missing options to chpst man page>
 Add options -t and -V to chpst man page; ignore -r since it's obsolete.
Author: <Lorenzo Puliti, Mike Pomraning 'Pilcrow'>
Origin: <Mike Pomraning 'Pilcrow'>, <https://github.com/pilcrow/runit>
Forwarded: <no>
Last-Update: 2022-01-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/doc/chpst.8.html
+++ b/doc/chpst.8.html
@@ -11,9 +11,9 @@
 <h2><a name='sect0'>Name</a></h2>
 chpst - runs a program with a changed process state 
 <h2><a name='sect1'>Synopsis</a></h2>
-<b>chpst</b> [-vP012]
+<b>chpst</b> [-vVP012]
 [-u <i>user</i>] [-U <i>user</i>] [-b <i>argv0</i>] [-e <i>dir</i>] [-/ <i>root</i>] [-n <i>inc</i>] [-l|-L <i>lock</i>] [-m <i>bytes</i>] [-d <i>bytes</i>]
-[-o <i>n</i>] [-p <i>n</i>] [-f <i>bytes</i>] [-c <i>bytes</i>] <i>prog</i> 
+[-o <i>n</i>] [-p <i>n</i>] [-f <i>bytes</i>] [-c <i>bytes</i>] [-t <i>seconds</i>] <i>prog</i>
 <h2><a name='sect2'>Description</a></h2>
 <i>prog</i> consists of one or
 more arguments. <p>
@@ -30,7 +30,7 @@
 of <i>user</i>&rsquo;s gid. If <i>group</i> consists of a colon-separated list of group names,
 <b>chpst</b> sets the group ids of all listed groups. If <i>user</i> is prefixed with
 a colon, the <i>user</i> and all <i>group</i> arguments are interpreted as uid and gids
-respectivly, and not looked up in the password or group file. All initial
+respectively, and not looked up in the password or group file. All initial
 supplementary groups are removed. </dd>
 
 <dt><b>-U <i>[:]user[:group]</b> </i></dt>
@@ -38,7 +38,7 @@
 variables $UID and $GID to the <i>user</i>&rsquo;s uid and gid, as found in <i>/etc/passwd</i>.
 If <i>user</i> is followed by a colon and a <i>group</i>, set $GID to the <i>group</i>&rsquo;s gid,
 as found in <i>/etc/group</i>, instead of <i>user</i>&rsquo;s gid. If <i>user</i> is prefixed with a
-colon, the <i>user</i> and <i>group</i> arguments are interpreted as uid and gid respectivly,
+colon, the <i>user</i> and <i>group</i> arguments are interpreted as uid and gid respectively,
 and not looked up in the password or group file. </dd>
 
 <dt><b>-b <i>argv0</b> </i></dt>
@@ -98,10 +98,17 @@
 <dd>limit core size. Limit the core
 file size to <i>bytes</i> bytes. </dd>
 
+<dt><b>-t <i>seconds</b> </i></dt>
+<dd>limit CPU time.  Limit CPU time to <i>seconds</i> seconds, delivering
+a SIGXCPU thereafter. </dd>
+
 <dt><b>-v</b> </dt>
 <dd>verbose. Print verbose messages to standard error.
 This includes warnings about limits unsupported by the system. </dd>
 
+<dt><b>-V</b> </dt>
+<dd>version string. Print a version string to standard error. </dd>
+
 <dt><b>-P</b> </dt>
 <dd>pgrphack.
 Run <i>prog</i> in a new process group. </dd>
--- a/man/chpst.8
+++ b/man/chpst.8
@@ -3,7 +3,7 @@
 chpst \- runs a program with a changed process state
 .SH SYNOPSIS
 .B chpst
-[\-vP012]
+[\-vVP012]
 [\-u
 .IR user ]
 [\-U
@@ -30,6 +30,8 @@
 .IR bytes ]
 [-c
 .IR bytes ]
+[-t
+.IR seconds ]
 .I prog
 .SH DESCRIPTION
 .I prog
@@ -68,7 +70,7 @@
 .I user
 and all
 .I group
-arguments are interpreted as uid and gids respectivly, and not looked up in
+arguments are interpreted as uid and gids respectively, and not looked up in
 the password or group file.
 All initial supplementary groups are removed.
 .TP
@@ -95,7 +97,7 @@
 .I user
 and
 .I group
-arguments are interpreted as uid and gid respectivly, and not looked up in
+arguments are interpreted as uid and gid respectively, and not looked up in
 the password or group file.
 .TP
 .B \-b \fIargv0
@@ -208,11 +210,21 @@
 .I bytes
 bytes.
 .TP
+.B \-t \fIseconds
+limit CPU time.
+Limit CPU time to
+.I seconds
+seconds, delivering a SIGXCPU thereafter.
+.TP
 .B \-v
 verbose.
 Print verbose messages to standard error.
 This includes warnings about limits unsupported by the system.
 .TP
+.B \-V
+version string.
+Print a version string to standard error.
+.TP
 .B \-P
 pgrphack.
 Run
--- a/src/chpst.c
+++ b/src/chpst.c
@@ -21,7 +21,7 @@
 #include "openreadclose.h"
 #include "direntry.h"
 
-#define USAGE_MAIN " [-vP012] [-u user[:group]] [-U user[:group]] [-b argv0] [-e dir] [-/ root] [-n nice] [-l|-L lock] [-m n] [-d n] [-o n] [-p n] [-f n] [-c n] prog"
+#define USAGE_MAIN " [-vVP012] [-u user[:group]] [-U user[:group]] [-b argv0] [-e dir] [-/ root] [-n nice] [-l|-L lock] [-m n] [-d n] [-o n] [-p n] [-f n] [-c n] [-t n] prog"
 #define FATAL "chpst: fatal: "
 #define WARNING "chpst: warning: "
 
--- a/src/chpst.dist
+++ b/src/chpst.dist
@@ -1,8 +1,8 @@
-usage: chpst [-vP012] [-u user[:group]] [-U user[:group]] [-b argv0] [-e dir] [-/ root] [-n nice] [-l|-L lock] [-m n] [-d n] [-o n] [-p n] [-f n] [-c n] prog
+usage: chpst [-vVP012] [-u user[:group]] [-U user[:group]] [-b argv0] [-e dir] [-/ root] [-n nice] [-l|-L lock] [-m n] [-d n] [-o n] [-p n] [-f n] [-c n] [-t n] prog
 
 100
 $Id: f279d44141c981dd7535a12260efcf1ef7beed26 $
-usage: chpst [-vP012] [-u user[:group]] [-U user[:group]] [-b argv0] [-e dir] [-/ root] [-n nice] [-l|-L lock] [-m n] [-d n] [-o n] [-p n] [-f n] [-c n] prog
+usage: chpst [-vVP012] [-u user[:group]] [-U user[:group]] [-b argv0] [-e dir] [-/ root] [-n nice] [-l|-L lock] [-m n] [-d n] [-o n] [-p n] [-f n] [-c n] [-t n] prog
 
 100
 test=1
